R/pkg/inst/tests/testthat/test_context.R

```diff
@@ -177,6 +177,13 @@ test_that("add and get file to be downloaded with Spark job on every node", {
   spark.addFile(path)
   download_path <- spark.getSparkFiles(filename)
   expect_equal(readLines(download_path), words)
+
+  # Test spark.getSparkFiles works well on executors.
+  seq <- seq(from = 1, to = 10, length.out = 5)
+  f <- function(seq) { readLines(spark.getSparkFiles(filename)) }
+  results <- spark.lapply(seq, f)
+  for (i in 1:5) { expect_equal(results[[i]], words) }
+
```

Comment:
The way you suggested can pass as well. After further investigating, I found [```install.spark()```](https://github.com/apache/spark/blob/master/R/pkg/tests/run-all.R#L25) is the root cause that makes my test fail. But I still not figure out which line in that function caused the failure.
I change the test case to a more simple way, which can still verify the fix. Without this fix, the updated test will fail. Since Spark 2.2 is coming and this fix passed for sparkR shell test and yarn cluster mode test, we can get this in firstly, and leave a todo for that weird test failure.